src/hooks/useFetch.ts

```diff
@@ -0,0 +1,39 @@
+import { useEffect, useState } from "react";
+
+interface ErrorBody {
+  status: number;
+  statusText: string;
+}
+
+export function useFetch<TResponse>(
+  url: string,
+  requestBody?: RequestInit,
+): { data: null | TResponse; isLoading: boolean; error: null | unknown } {
+  const [data, setData] = useState<null | TResponse>(null);
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState<ErrorBody | null>(null);
+
+  useEffect(() => {
+    async function makeRequest(): Promise<void> {
+      setIsLoading(true);
+
+      const response: Response = await fetch(url, requestBody);
+
+      // Set an error if the request ins't successful
+      if (response.status < 200 || response.status > 299) {
+        setError({ status: response.status, statusText: response.statusText });
+        setIsLoading(false);
+        return;
+      }
+
+      const jsonResponseBody: TResponse = await response.json();
+
+      setData(jsonResponseBody);
+      setIsLoading(false);
+    }
```

Comment:
We can extract this function later to execute fetch on demand like ticket-update, which can be difficult to manage with hooks. But I'll leave this function here until we get to work on that scenario.